File: 367_perfect_square.c
bool isPerfectSquare(int num)
{
    int left = 0;
    int right = num;
    while (left <= right)
    {
        long int middle = left + (right - left) / 2;
        long int sqr = middle * middle;
        if (sqr < num)
        {
            left = middle + 1;
        }
        else if (sqr > num)
        {
            right = middle - 1;
        }
        else
        {
            return true;
        }
        
    }
    return false;
    
}

File: 367_perfect_square.py
class Solution:
    def isPerfectSquare(self,num: int) -> bool:
        left , right = 0, num
        while left <= right:
            middle = left + (right - left) // 2
            sqr = middle * middle
            if(sqr < num):
                left = middle + 1
            elif(sqr > num):
                right = middle -1
            else:
                return True

        return False